pkg/providers/provider.go

```diff
@@ -0,0 +1,30 @@
+package provider
+
+import (
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+// Provider defines the methods required by source providers for importing a VM
+type Provider interface {
+	Connect(*corev1.Secret) error
+	Close()
+	SetVM(v2vv1alpha1.VirtualMachineImportSourceSpec) error
```

Comment:
What do you think about calling it more provider-like, i.e. `LoadVM` or `FetchVM`?